src/reducer.js

```diff
@@ -0,0 +1,36 @@
+const initialState = {
+  cats: []
+}
+
+function setState (state, mewState) {
+  return Object.assign({}, state, mewState)
+}
+
+function removeCat (state, id) {
+  return Object.assign({}, state, {
+    cats: state.cats.filter(cat => cat.id !== id)
+  })
+}
+
+function fetchCatsRequest (state) {
+  return Object.assign({}, state)
+}
+
+function fetchCatsSuccess (state, cats) {
+  return Object.assign({}, state, {cats})
+}
+
+export default function rootReducer (state = initialState, action) {
+  switch (action.type) {
+  case "SET_STATE":
+    return setState(state, action.state)
+  case "REMOVE_CAT":
+    return removeCat(state, action.id)
+  case "FETCH_CATS_REQUEST":
```

Comment:
this is where you would do your optimistic update if you were to have one - so you would update your state with user entered date, keep a ref to the old state, on success, replace the state or on fail revert the state